src/sentry/api/base.py

```diff
@@ -246,6 +246,25 @@ def paginate(
         return response
 
 
+class EnvironmentMixin(object):
+    def _get_environment_id_from_request(self, request, organization_id):
+        environment = self._get_environment_from_request(request, organization_id)
+
+        return environment and environment.id
+
+    def _get_environment_from_request(self, request, organization_id):
+        # TODO: cache env on request to avoid hitting memcached over and over
```

Comment:
Any reason not to just do this right now?

Reply:
Nope, missed that TODO, thanks.